proposals/lip-Introduce_PoA_module.md

```diff
@@ -0,0 +1,607 @@
+Its initial value is set in the genesis block.
+
+#### Chain Properties Substore
+
+This substore contains the general properties of the chain.
+
+##### Store Prefix, Store Key, and Store Value
+
+* The store prefix is set to `STORE_PREFIX_CHAIN`.
+* The store key is set to empty bytes.
+* The store value is set to the serialization of an object following `chainPropSchema` below.
+* Notation: Let `chainProperties` be the entry in the chain properties substore.
+
+##### JSON Schema
+
+```java
+chainPropSchema = {
+   "type": "object",
+   "properties": {
+      "roundEndHeight": {
+          "dataType": "uint32",
+          "fieldNumber": 1
+      },
+       "validatorsUpdateNonce": {
+           "dataType": "uint32",
+           "fieldNumber": 2
+      },
+    }
+    "required": [
+       "roundEndHeight",
+       "validatorsUpdateNonce"
+    ]
+}
+```
+
+##### Properties and Default Values
+
+The properties of this schema are as follows:
+
+*   `roundEndHeight`: An integer stating the last height of the round. 
+Its initial value is set after the execution of the genesis block.
+*   `validatorsUpdateNonce`: An integer representing the number of times that the validator set has been updated with an update auhtority command. 
+It is initialized to 0.
+
+### Commands
+
+#### Authority Registration Command
+
+This command is equivalent to the [delegate registration command][delegatedoc] in the DPoS module and has the same schema and similar validity rules. 
+The command ID of this transaction is `COMMAND_ID_REGISTRATION_AUTHORITY`.
+
+##### Parameters
+
+```java
+registrationTransactionParamsSchema = {
+  "type": "object",
+  "properties": {
+    "name": {
+      "dataType": "string",
+      "fieldNumber": 1
+    },
+    "blsKey": {
+      "dataType": "bytes",
+      "fieldNumber": 2
+    },
+    "proofOfPossession": {
+      "dataType": "bytes",
+      "fieldNumber": 3
+    },
+    "generatorKey": {
+      "dataType": "bytes",
+      "fieldNumber": 4
+    }
+  },
+  "required": ["name", "blsKey", "proofOfPossession", "generatorKey"]
+}
+```
+
+##### Verification 
+ 
+Let `trs` be a transaction with module ID `MODULE_ID_POA` and command ID `COMMAND_ID_REGISTRATION_AUTHORITY` to be verified. 
+The list of verification conditions for `trs.params` is as follows:
+
+* The `trs.params.name` property has to contain only characters from the set `[a-z0-9!@$&_.]`, must not be empty and has to be at most `MAX_LENGTH_NAME` characters long.
+* Let `address` be the 20-byte address derived from `trs.senderPublicKey`. Then `address` must not already be registered as a validator. This is, the validator substore has no entry with store key `address`.
+* The value of `trs.params.name` must not already be registered as a validator name. This is, the name substore has no entry with store key `trs.params.name`.
+* `isKeyRegistered(trs.params.blsKey)` must return `true`, where the function is defined in the [Validators module][validatorsModule].
```

Comment:
Shouldn't `isKeyRegistered(trs.params.blsKey)` return `False`?

Reply:
yes, it should, the BLS key can't exist in the blockchain for this command to be valid. Good catch.